### DISCUSSION TITLE
Analyzing the trace file generated by z3

### DISCUSSION THREAD

**Discussion by rahxephon89** [Z3Prover/z3#6874] [Q&A] [answered]
Hi, I am trying to analyze performance of quantifier instantiation by parsing the trace file generated by z3. However, the [z3tracer](https://github.com/facebookarchive/smt2utils/tree/dc24b2ba983b657af7c1641d15cecc75cf739f3c/z3tracer) tool only supports z3 4.8.9. I am wondering whether there is a reference doc for parsing the generated trace file. Thanks!

**Comment by NikolajBjorner** [comment, marked as answer]
The legacy tools is AxiomProfiler, https://github.com/viperproject/axiom-profiler.
Maybe @alexanderjsummers has updates.

You could also try custom scripts around the proof logs:
https://microsoft.github.io/z3guide/programming/Proof%20Logs

You could also log quantifier instantiations by setting a configuration:
z3 solver.instantiations2console=true file.smt2
